staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -893,3 +886,101 @@ func setObjectMeta(u *unstructured.Unstructured, objectMeta *metav1.ObjectMeta)
 	u.UnstructuredContent()["metadata"] = metadata
 	return nil
 }
+
+type implicitField struct {
+	path   string
+	exists bool
+	value  interface{}
+}
+
+func getImplicitFields(u *unstructured.Unstructured) ([]implicitField, error) {
```

Comment:
I want to get an initial review on this before adding all fields in objectMeta here.

Reply:
And if we end up setting these fields explicitly, do we still want to round trip through marshalling and unmarshalling or go field by field accumulating into the metadata object by iterating through the `metadataMap`?

Reply:
ref https://github.com/kubernetes/kubernetes/pull/64267#discussion_r190924277

Reply:
I don't think this is the right layer at which to do this. I think the unstructured objectmeta accessor implementations should be adjusted to produce an unstructured object that matches ObjectMeta omitempty semantics

Reply:
For example, if SetClusterName("") is called, that should remove the clusterName element from the unstructured object

Reply:
tricky one: do we need to do cascading element removal in accessor for omitempty semantics?

Reply:
I think all the accessor methods are for top level fields in metadata, so I'm not sure that's an issue

Reply:
updated.

Reply:
> I think the unstructured objectmeta accessor implementations should be adjusted to produce an unstructured object that matches ObjectMeta omitempty semantics

The accessors were already modified to support this for some of the fields. Added for the rest.